packages/arb-validator-core/message/message.go

```diff
@@ -34,20 +34,26 @@ const (
 	ERC721Type
 	ContractTransactionType
 	CallType
+	TransactionBatchType
 )
 
-type UnsentMessage interface {
-	Type() MessageType
-	AsValue() value.Value
-	GetFuncName() string
+type SingleMessage interface {
+	ReceiptHash() common.Hash
+
+	asValue() value.Value
+	deliveredHeight() *common.TimeBlocks
+	deliveredTimestamp() *big.Int
 }
 
 type Message interface {
-	UnsentMessage
+	fmt.Stringer
+	Type() MessageType
 	Equals(o Message) bool
-	ReceiptHash() common.Hash
-	DeliveredHeight() *common.TimeBlocks
-	DeliveredTimestamp() *big.Int
+}
+
+type ExecutionMessage interface {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [exported type ExecutionMessage should have comment or be unexported](https://app.codacy.com/gh/OffchainLabs/arbitrum/pullRequest?prid=5428340)